aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -358,6 +367,40 @@ impl VMChangeSet {
         Ok(())
     }
 
+    fn squash_additional_resource_writes(
+        write_set: &mut HashMap<StateKey, (WriteOp, Option<MoveTypeLayout>)>,
+        additional_write_set: HashMap<StateKey, (WriteOp, Option<MoveTypeLayout>)>,
+    ) -> anyhow::Result<(), VMStatus> {
+        for (key, additional_entry) in additional_write_set.into_iter() {
+            match write_set.entry(key) {
+                Occupied(mut entry) => {
+                    // Squash entry and addtional entries if type layouts match
+                    let (additional_write_op, additional_type_layout) = additional_entry;
+                    let (write_op, type_layout) = entry.get_mut();
+                    if *type_layout != additional_type_layout {
```

Comment:
Sorry, confusing wording, I meant traversal in general. I think we do not really need to copy type layouts, we either own them or we need a reference. I would really love to avoid traversals for equality, otherwise this can be expensive and hurt performance.

Reply:
Let's add TODO here, and then remove the check after everything is working. 

this seems like a good sanity while developing, but can just remove later